src/main/java/com/openlattice/datastore/configuration/DatastoreConfiguration.kt

```diff
@@ -0,0 +1,27 @@
+package com.openlattice.datastore.configuration
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.kryptnostic.rhizome.configuration.Configuration;
+import com.kryptnostic.rhizome.configuration.ConfigurationKey
+import com.kryptnostic.rhizome.configuration.SimpleConfigurationKey
+import com.kryptnostic.rhizome.configuration.annotation.ReloadableConfiguration
+import com.openlattice.client.serialization.SerializationConstants
+
+@ReloadableConfiguration(uri = "datastore.yaml")
+data class DatastoreConfiguration(
+        @JsonProperty("bucketName") val bucketName: String,
+        @JsonProperty("regionName") val regionName: String,
+        @JsonProperty("timeToLive") val timeToLive: Long,
+        @JsonProperty("accessKeyId") val accessKeyId: String,
+        @JsonProperty("secretAccessKey") val secretAccessKey: String): Configuration {
+    val key = SimpleConfigurationKey("datastore.yaml")
+
+    @JsonIgnore
+    override fun getKey(): ConfigurationKey {
+        return key
+    }
+
+    @JsonIgnore
+    fun key(): ConfigurationKey { return key }
```

Comment:
This will NPE and you don't need the @JsonIgnore.  Check out LinkingConfiguration in indexer for the correct pattern.

Reply:
Where/when will the NPE occur? I've been able to load all these values and use them when needed in the local aws and local blob data services.

Reply:
I added the companion object like in LinkingConfiguration and it works for the bamboo test so I think we should be good on this one